distributions/raspijamulus.sh

```diff
@@ -1,16 +1,19 @@
 #!/bin/bash
 
-# This script is intended to setup a clean Raspberry Pi system for running Jamulus
-OPUS="opus-1.3.1"
+# This script is intended to setup a clean Raspberry Pi system for running Jamulus.
+# Please run this script from the distributions/ folder
+
+readonly OPUS="opus-1.3.1"
 NCORES=$(nproc)
```

Comment:
You can say `declare -ri NCORES="$(nproc)"` to declare NCORES both read-only _and_ containing an integer value.

Reply:
I did that but spellcheck complained: https://www.shellcheck.net/

Reply:
Oh dear...  What did it complain about?

Reply:
```
declare -ri NCORES=$(nproc)
            ^-- [SC2155](https://github.com/koalaman/shellcheck/wiki/SC2155) (warning): Declare and assign separately to avoid masking return values.
```

Reply:
Oh dear, then it is pretty broken.  You _cannot_ declare something readonly and then assign to it...  It's right about assignment masking the exit code:
```
~$ ( declare -ri NCORES=$(nproc) || { echo "foo"; exit 1; }; echo $NCORES )
6
~$ ( declare -ri NCORES=$(barbaz) || { echo "foo"; exit 1; }; echo $NCORES )
barbaz: command not found
0
```
Where the second go should have said "foo", not "0".  So it should equally not like
```
readonly NCORES=$(nproc)
```
on that basis:
```
$ ( readonly NCORES=$(barbaz) || { echo "foo"; exit 1; }; echo $NCORES )
barbaz: command not found

$
```
(no "foo" echoed).  You end up with
```
tempvarnamehere=$(ncores) || exit 1
declare -ri NCORES="$tempvarnamehere"
```
which should keep it happy.  Otherwise you end up having `"...${NCORES}..."` everywhere.